Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1072,6 +1473,608 @@ def get_cim_mapping_field_command():
+    )
+    return datetime.utcnow() - earliest_job_datetime > timedelta(minutes=enrichment_timeout)
+
+
+def reset_enriching_fetch_mechanism():
+    """ Resets all the fields regarding the enriching fetch mechanism & the last run object """
+
+    integration_context = get_integration_context()
+    for field in (INCIDENTS, CACHE):
+        if field in integration_context:
+            del integration_context[field]
+    set_integration_context(integration_context)
+    demisto.setLastRun({})
+    demisto.results("Enriching fetch mechanism was reset successfully.")
+
+
+def fetch_incidents(service, enrichment_timeout, num_enrichment_events):
+    if ENABLED_ENRICHMENTS:
+        integration_context = get_integration_context()
+        if not demisto.getLastRun() and integration_context:
+            # In "Pull from instance" in Classification & Mapping the last run object is empty, integration context
+            # will not be empty because of the enrichment mechanism. In regular enriched fetch, we use dummy data
+            # in the last run object to avoid entering this case
+            fetch_incidents_for_mapping(integration_context)
+        else:
+            run_enrichment_mechanism(service, enrichment_timeout, integration_context, num_enrichment_events)
+    else:
+        fetch_notables(service=service, enrich_notables=False)
+
+
+def run_enrichment_mechanism(service, enrichment_timeout, integration_context, num_enrichment_events):
+    """ Execute the enriching fetch mechanism
+
+    Args:
+        service (splunklib.client.Service): Splunk service object.
+        enrichment_timeout (int): The timeout for the enrichment process.
+        Possible values are: Drilldown, Asset & Identity.
+        integration_context (dict): The integration context
+        num_enrichment_events (int): The maximal number of events to return per enrichment type.
+
+    """
+    incidents = []  # type: list
+    cache_object = Cache.from_json(json.loads(integration_context.get(CACHE, "{}")))  # Cache deserialization
+
+    try:
+        done_handling = handle_open_enrichments(service, enrichment_timeout, incidents, cache_object)
+        done_enriching = is_done_enriching(cache_object)
+        if done_handling and done_enriching:
+            fetch_notables(service=service, cache_object=cache_object, enrich_notables=True)
+        enrich_fetched_notables(service, incidents, num_enrichment_events, cache_object)
+
+    except Exception as e:
+        err = 'Caught an exception while executing the enriching fetch mechanism. Additional Info: {}'.format(str(e))
+        demisto.error(err)
+        raise e
+
+    finally:
+        store_incidents_for_mapping(incidents, integration_context)
+        integration_context[CACHE] = json.dumps(cache_object, default=lambda obj: obj.__dict__)  # Cache serialization
+        set_integration_context(integration_context)  # Saving the integration context for the next run
```

Comment:
Same as above, Let's add a function to the Cache class to do this, we'll call it something like save_cache_to_context, it will do exactly this ☝️